x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/get_filter.test.ts

```diff
@@ -0,0 +1,431 @@
+      );
+      expect(filter).toEqual({
+        bool: {
+          must: [],
+          filter: [
+            {
+              bool: {
+                should: [
+                  {
+                    match: {
+                      'host.name': 'siem',
+                    },
+                  },
+                ],
+                minimum_should_match: 1,
+              },
+            },
+          ],
+          should: [],
+          must_not: [],
+        },
+      });
+    });
+
+    test('throws on type query if query is undefined', async () => {
+      await expect(
+        getFilter(
+          'query',
+          undefined,
+          undefined,
+          undefined,
+          'host.name: siem',
+          undefined,
+          servicesMock,
+          ['auditbeat-*']
+        )
+      ).rejects.toThrow('query, filters, and index parameter should be defined');
+    });
+
+    test('throws on type query if language is undefined', async () => {
+      await expect(
+        getFilter('query', undefined, undefined, 'kuery', undefined, undefined, servicesMock, [
+          'auditbeat-*',
+        ])
+      ).rejects.toThrow('query, filters, and index parameter should be defined');
+    });
+
+    test('throws on type query if index is undefined', async () => {
+      await expect(
+        getFilter(
+          'query',
+          undefined,
+          undefined,
+          'kuery',
+          'host.name: siem',
+          undefined,
+          servicesMock,
+          undefined
+        )
+      ).rejects.toThrow('query, filters, and index parameter should be defined');
+    });
+
+    test('returns a saved query if given a type of query', async () => {
+      const filter = await getFilter(
+        'saved_query',
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        'some-id',
+        servicesMock,
+        ['auditbeat-*']
+      );
+      expect(filter).toEqual({
+        bool: {
+          filter: [
+            { bool: { minimum_should_match: 1, should: [{ match: { 'host.name': 'linux' } }] } },
+          ],
+          must: [],
+          must_not: [],
+          should: [],
+        },
+      });
+    });
+
+    test('throws on saved query if saved_id is undefined', async () => {
+      await expect(
+        getFilter(
+          'saved_query',
+          undefined,
+          undefined,
+          undefined,
+          undefined,
+          undefined,
+          servicesMock,
+          ['auditbeat-*']
+        )
+      ).rejects.toThrow('savedId parameter should be defined');
+    });
+
+    test('throws on saved query if index is undefined', async () => {
+      await expect(
+        getFilter(
+          'saved_query',
+          undefined,
+          undefined,
+          undefined,
+          undefined,
+          'some-id',
+          servicesMock,
+          undefined
+        )
+      ).rejects.toThrow('savedId parameter should be defined');
+    });
+  });
+});
```

Comment:
Expand this file if github is trying to hide it as this is all important tests for the system.

Reply:
nice tests 👍